CRM/Price/BAO/LineItem.php

```diff
@@ -1282,7 +1282,7 @@ protected function addFinancialItemsOnLineItemChange($isCreateAdditionalFinancia
     // insert financial items
     // ensure entity_financial_trxn table has a linking of it.
     CRM_Financial_BAO_FinancialItem::add($lineObj, $updatedContribution, NULL, $tempFinancialTrxnID);
-    if (isset($lineObj->tax_amount)) {
+    if (isset($lineObj->tax_amount) && $lineObj->tax_amount !== '0.00') {
```

Comment:
@monishdeb @pradpnayak are you OK with this change as a test -fix - see comments in commit message ^^

Reply:
I am not 100% sure for this change. What will happen if line item had a tax and then changed it 0? Will this function be invoked and create a negative entry for old tax financial item entry?
@monishdeb do you have any idea?